oioioi/contests/utils.py

```diff
@@ -247,6 +256,96 @@ def visible_rounds(request):
     return [r for r in queryset if controller.can_see_round(request, r)]
 
 
+@request_cached
+def get_number_of_rounds(request):
+    """Returns the number of rounds in the current contest.
+    """ 
+    return Round.objects.filter(contest=request.contest).count()
+
+
+def get_contest_dates(request):
+    """Returns the end_date of the latest round and the start_date
+    of the earliest round.
+    """ 
+    if isinstance(request, HttpRequest):
+        rtimes = rounds_times(request, request.contest)
+    else:
+        rtimes = generic_rounds_times(None, request.contest)
```

Comment:
Can you explain the logic here? To me, the other branch never happens.

Reply:
You're right, corrected.